Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -12835,8 +12848,324 @@ def pan_os_delete_application_group_command(args):
+    """get the Job ID linked to a particular query.
+
+    Args:
+        log_type (str): query log type
+        query (str): query for the fetch
+        max_fetch (int): maximum number of entries to fetch
+
+    Returns:
+        job_id (str): returns the Job ID associated with the given query
+    """
+    params = assign_params(key=API_KEY, type='log', log_type=log_type.lower(), query=query, nlogs=max_fetch)
+    response = http_request(URL, 'GET', params=params)
+    job_id = response.get('response', {}).get('result', {}).get('job')
+    return job_id
+
+
+def get_query_entries_by_id_request(job_id: str) -> Dict[str, Any]:
+    """get the entries of a particular Job ID.
+
+    Args:
+        job_id (int): ID of a query job
+
+    Returns:
+        Dict[str,Any]: a dictionary of the raw entries linked to the Job ID
+    """
+    params = assign_params(key=API_KEY, type='log', action='get', job_id=job_id)
+    response = http_request(URL, 'GET', params=params)
+    return response
+
+
+def get_query_entries(log_type: str, query: str, max_fetch: int) -> List[Dict[Any, Any]]:
+    """get query entries according to a specific query.
+    
+    Args:
+        log_type (str): query log type
+        query (str): query for the fetch
+        max_fetch (int): maximum number of entries to fetch
+
+    Returns:
+        List[Dict[Any,Any]]): a list of raw entries for the the specified query
+    """
+    entries = []
+    # first http request: send request with query, valid response will contain a job id.
+    job_id = get_query_job_id_request(log_type, query, max_fetch)
+    demisto.debug(f'{job_id=}')
+
+    # second http request: send request with job id, valid response will contain a dictionary of entries.
+    query_entries = get_query_entries_by_id_request(job_id)
+
+    # extract all entries from response
+    if result := query_entries.get('response', {}).get('result', {}).get('log', {}).get('logs', {}).get('entry'):
+        if isinstance(result,list):
+            entries.extend(result)
+        elif isinstance(result,dict):
+            entries.append(result)
+        else:
+            raise DemistoException('Could not parse fetch results.')
+    
+    entries_log_info = {entry.get('seqno',''):entry.get('time_generated') for entry in entries}
+    demisto.debug(f'{log_type} log type: {len(entries)} raw incidents (entries) found.')
+    demisto.debug(f'fetched raw incidents (entries) are (ID:time_generated): {entries_log_info}')
+    return entries
 
-def main():
+
+def add_time_filter_to_query_parameter(query: str, last_fetch: datetime) -> str:
+    """append time filter parameter to original query parameter.
+
+    Args:
+        query (str): a string representing a query
+        last_fetch (datetime): last fetch time for the specific log type
+
+    Returns:
+        str: a string representing a query with added time filter parameter
+    """
+    if 'time_generated' in query or not last_fetch:
+        return query
```

Comment:
raising as we do not allow user to mess with this field as we depend on it.
and lets add it to test-module.